cupy/testing/_array.py

```diff
@@ -85,12 +86,15 @@ def assert_array_equal(x, y, err_msg='', verbose=True, strides_check=False):
          err_msg(str): The error message to be printed in case of failure.
          verbose(bool): If ``True``, the conflicting values
              are appended to the error message.
+         strict(bool): If ``True``, raise an AssertionError when either
+             the shape or the data type of the array_like objects does not
+             match.
 
     .. seealso:: :func:`numpy.testing.assert_array_equal`
     """
     numpy.testing.assert_array_equal(
         cupy.asnumpy(x), cupy.asnumpy(y), err_msg=err_msg,
-        verbose=verbose)
+        verbose=verbose, strict=strict)
```

Comment:
This will make CuPy incompatible with NumPy 1.23 or earlier. I think it's better to receive kwargs and pass-thorugh it to `assert_array_equal`.

Reply:
I have modified it so that now the function accepts kwargs. Modified docstring in `strict` option to explicitly state the required NumPy version.